cirq-core/cirq/transformers/target_gatesets/compilation_target_gateset.py

```diff
@@ -115,3 +115,95 @@ def postprocess_transformers(self) -> List['cirq.TRANSFORMER']:
             drop_negligible.drop_negligible_operations,
             drop_empty_moments.drop_empty_moments,
         ]
+
+
+class TwoQubitCompilationTargetGateset(CompilationTargetGateset):
+    """Abstract base class to create two-qubit target gatesets.
+
+    This base class can be used to create two-qubit compilation target gatesets. It automatically
+    implements the logic to
+
+        1. Expand composite gates acting on > 2 qubits using
+        2. Merge connected components of 1 and 2 qubit unitaries into tagged circuit operations.
+        3. Use `_decompose_single_qubit_operation` and `_decompose_two_qubit_operation` to figure
+        out how to rewrite the merged connected components using gates from this gateset.
+        4. Replace connected components with inefficient implementations (higher number of 2q
+        interactions) with more efficient rewrites to minimize total number of 2q interactions.
+        5. Apply cleanup transformations to preserve moment structure (eg: alternating layers of
+        single/two qubit gates) and minimize circuit depth (eg: drop negligible operations and
+        empty moments)
```

Comment:
Nit: can we name the two class properties here for `pre` and `post` transformers here so the user knows what to go and look at (Might be good to do in steps 1 and 2 as well).

Reply:
done